lib/maze/request_repeater.rb

```diff
@@ -0,0 +1,41 @@
+module Maze
+  # Repeats POST requests
+  class RequestRepeater
+
+    # @param request_type [String] The type of request being handled by this repeater
+    def initialize(request_type)
+      @request_type = request_type
+    end
+
+    # @param request [HTTPRequest] The request to be repeated
+    def repeat(request)
+
+      # TODO: Forwarding of internal errors to be considered later
+      return if request.header.keys.any? { |key| key.downcase == 'bugsnag-internal-error' }
+
+      url = URI.parse(url_for_request_type)
+      http = Net::HTTP.new(url.host)
+      bugsnag_request = Net::HTTP::Post.new(url.path)
+      bugsnag_request['Content-Type'] = 'application/json'
+
+      # Set all Bugsnag headers that are present
+      bugsnag_request.body = request.body
+      request.header.each {|key,value| bugsnag_request[key] = value if key.downcase.start_with? 'bugsnag-' }
+      bugsnag_request['bugsnag-api-key'] = Maze.config.repeater_api_key
+
+      # TODO Also overwrite apiKey in the payload, if present, and recalculate the integrity header
+
+      http.request(bugsnag_request)
+    end
+
+    private
+
+    def url_for_request_type
```

Comment:
Logic now implemented here based on request_type, rather than passing it down.  I ran into trouble trying the `prepend` approach, as `do_POST` exists at two level and both need be be run.  I'm sure it's possible, but this felt the most easily achievable and better than it was before.

Reply:
Calling `super` in a prepended module will call the class' method:

```ruby
module A
  def x
    puts "A#x"
    super
  end
end

class B
  prepend A
  
  def x
    puts "B#x"
  end
end

B.new.x

# => A#x
# => B#x
```

That should be all you need to do to run both methods